netinterface.py

```diff
@@ -184,20 +191,44 @@ def writeRHStyleInterface(self, iface):
             f.write("NETMASK=%s\n" % self.netmask)
             if self.gateway:
                 f.write("GATEWAY=%s\n" % self.gateway)
+
+        if self.modev6:
+            with open('/etc/sysconfig/network', 'w') as net_conf:
+                net_conf.write("NETWORKING_IPV6=yes\n")
+            f.write("IPV6INIT=yes\n")
+            f.write("IPV6_DEFROUTE=yes\n")
+            f.write("IPV6_DEFAULTDEV=%s\n" % iface_vlan)
+            if self.modev6 == self.DHCP:
+                f.write("DHCPV6C=yes\n")
+                f.write("PERSISTENT_DHCLIENT_IPV6=yes\n")
+                f.write("IPV6_FORCE_ACCEPT_RA=yes\n")
+                f.write("IPV6_AUTOCONF=no\n")
+            elif self.modev6 == self.Static:
+                f.write("IPV6ADDR=%s\n" % self.ipv6addr)
+                if self.ipv6_gateway:
+                    f.write("IPV6_DEFAULTGW=%s\n" % (self.ipv6_gateway))
+                f.write("IPV6_AUTOCONF=no\n")
+            elif self.modev6 == self.Autoconf:
+                f.write("IPV6_AUTOCONF=yes\n")
+
         if self.vlan:
             f.write("VLAN=yes\n")
         f.close()
 
 
     def waitUntilUp(self, iface):
-        if not self.isStatic():
-            return True
-        if not self.gateway:
-            return True
+        iface_name = self.getInterfaceName(iface)
+        if self.isStatic4() and self.gateway and util.runCmd2(
+            ['/usr/sbin/arping', '-f', '-w', '120', '-I', iface_name, self.gateway]
+        ):
+            return False
+
+        if self.isStatic6() and self.ipv6_gateway and util.runCmd2(
+            ['/usr/sbin/ndisc6', '-1', '-w', '120', self.ipv6_gateway, iface_name]
+        ):
+            return False
```

Comment:
This code is equivalent to the `arping` code above but for IPv6 as `arping` only supports IPv4.
Please note this requires to have `ndisc6` installed in host-installer image.

Reply:
Is this reflected in the corresponding RPM spec file?

Reply:
As far as I know your installer spec file repository is not public so I can't provide a PR to add `ndisc6` which is why I commented my code for someone with access to reflect the change. :)